contracts/collect/AaveFeeCollectModule.sol

```diff
@@ -0,0 +1,258 @@
+// SPDX-License-Identifier: AGPL-3.0-only
+
+pragma solidity 0.8.10;
+
+import {Errors} from '@aave/lens-protocol/contracts/libraries/Errors.sol';
+import {FeeModuleBase} from '@aave/lens-protocol/contracts/core/modules/FeeModuleBase.sol';
+import {ICollectModule} from '@aave/lens-protocol/contracts/interfaces/ICollectModule.sol';
+import {ModuleBase} from '@aave/lens-protocol/contracts/core/modules/ModuleBase.sol';
+import {FollowValidationModuleBase} from '@aave/lens-protocol/contracts/core/modules/FollowValidationModuleBase.sol';
+
+import {IPoolAddressesProvider} from '@aave/core-v3/contracts/interfaces/IPoolAddressesProvider.sol';
+import {IPoolDataProvider} from '../interfaces/IPoolDataProvider.sol';
+import {IPool} from '@aave/core-v3/contracts/interfaces/IPool.sol';
+
+import {EIP712} from '@openzeppelin/contracts/utils/cryptography/draft-EIP712.sol';
+import {IERC20} from '@openzeppelin/contracts/token/ERC20/IERC20.sol';
+import {IERC721} from '@openzeppelin/contracts/token/ERC721/IERC721.sol';
+import {SafeERC20} from '@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol';
+
+/**
+ * @notice A struct containing the necessary data to execute collect actions on a publication.
+ *
+ * @param collectLimit The maximum number of collects for this publication. 0 for no limit.
+ * @param currentCollects The current number of collects for this publication.
+ * @param amount The collecting cost associated with this publication.
+ * @param currency The currency associated with this publication.
+ * @param recipient The recipient address associated with this publication.
+ * @param referralFee The referral fee associated with this publication.
+ * @param endTimestamp The end timestamp after which collecting is impossible. 0 for no expiry.
+ */
+struct ProfilePublicationData {
+    uint256 collectLimit;
+    uint256 currentCollects;
+    uint256 amount;
+    address currency;
+    address recipient;
+    uint16 referralFee;
+    bool followerOnly;
+    uint40 endTimestamp;
+}
```

Comment:
I was wondering if considering this:
```solidity
struct ProfilePublicationData {
    ///////////////////////// Slot 0
    uint256 collectLimit;
    
    ///////////////////////// Slot 1
    uint256 currentCollects;
    
    ///////////////////////// Slot 2
    uint256 amount;
    
    ///////////////////////// Slot 3
    address currency;
    // <96 unused bits>
    
    ///////////////////////// Slot 4
    address recipient;
    uint16 referralFee;
    bool followerOnly;
    uint40 endTimestamp;
    // <32 unused bits>
}
```

Might worth to move from slot and increase the bits of the timestamp, to allow higher timestamps without any increase in storage cost:
```solidity
struct ProfilePublicationData {
    ///////////////////////// Slot 0
    uint256 collectLimit;
    
    ///////////////////////// Slot 1
    uint256 currentCollects;
    
    ///////////////////////// Slot 2
    uint256 amount;
    
    ///////////////////////// Slot 3
    address currency;
    uint96 endTimestamp;
    
    ///////////////////////// Slot 4
    address recipient;
    uint16 referralFee;
    bool followerOnly;
    // <72 unused bits>
}
```

Also another storage packing might be possible if we put a cap in the collect limit to 2^96 which is 79,228,162,514,264,337,593,543,950,336 (looks high enough!)
```solidity
struct ProfilePublicationData {
    ///////////////////////// Slot 0
    uint256 amount;
    
    ///////////////////////// Slot 1
    address currency;
    uint96 collectLimit;
    
    ///////////////////////// Slot 2
    uint96 currentCollects;
    address recipient;

    ///////////////////////// Slot 3
    uint16 referralFee;
    bool followerOnly;
    uint72 endTimestamp;
}
```

What do you think?

Reply:
Nice struct packing ser, I like the last one